Server testing

ghuser1023 opened the pull request:
Adds testing via Mocha (https://mochajs.org/), akin to what's in https://socket.io/docs/v3/testing/. This enables us to:
- Write integration tests that involve actual messages sent over sockets, like in `tests\server.test.ts` below
- Write unit tests that test individual functionality (not yet defined)

This was chosen over mocking via https://www.npmjs.com/package/socket.io-mock for the following reasons:
- We don't actually want to mock out the sockets, since we want to make sure that they're working.
- Mocking is complex under the hood (not reflected in the simplicity of the provided example), and hurts my brain.

Changes of note:
- Moved all server logic from `index.ts` -> `server.ts`, leaving just the initialization in `index.ts`. This lets us write test code that tests the actual server implementation.
- A simple test (under `tests/`) testing the current server functionality -- this should be rewritten once the server starts to actually do more things.
- New autodoc generation command, reflecting the existence of `server.ts`.



ADDITIONAL:
- Update heroku api key to allow continous deployment to work properly

ghuser1023 committed:
add test dependencies

ghuser1023 committed:
additional typescript dependencies

ghuser1023 committed:
move server to server.ts

ghuser1023 committed:
add test file

ghuser1023 committed:
modify autodoc generation to accomodate refactoring

ghuser1023 committed:
remove unnecessary unix advisory

reginawang3495 commented on server/README.md:
```diff
@@ -23,9 +23,13 @@
     - Currently, RemoteController.cs is attached to the RemoteController GameObject
 
```
at like line 17 we can change to server.ts if you want, ow i can do on my own next commit!

reginawang3495 reviewed (dismissed):
noice :)

reginawang3495 commented:
oh also do we wanna maybe add a setup testing task to #17 or something?

ghuser1023 committed:
modify readme

ghuser1023 committed:
Merge branch 'master' into server-test

ghuser1023 commented:
> at like line 17 we can change to server.ts if you want, ow i can do on my own next commit!

fixed

> oh also do we wanna maybe add a setup testing task to #17 or something?

added a bullet point! don't quite feel like making a task haha

reginawang3495 committed:
Add server CICD to pipeline (#65)

* add cicd

* change nodemon prod to tsnode prod

* try fix build

* add tsc

* fix require error

* fix tsc

* remove

* npm install
:

* heroku fix

* heroku fix

* typescript import

* run compiled js

* put back command

* accidentally removed tsc command

* port to env port if given

* add docs

* fix typo

* update config site link

markgue committed:
Basic player movement (#66)

* Basic scene and framework for robot implementation

* More skeleton code in robot movement

Added helper Direction enum and function that will be helpful later

* Implemented robot movement

Demo is in MarkWorkspace. Use WASD to move. Currently a little buggy since the program won't be expecting async inputs from the player when fully implemented later. Tap WASD once and wait until it completes moving.

* Start of BoardManager

Can generate a board based on width/height

* Create a board of arbitrary size

* Places players at opposite corners

Player 1 in bottom left, Player 2 in top right

* Added doxygen plugin to Unity + generated documentation

Doxygen plugin allows us to generate Doxygen documentation for the entire Unity project easily.

* Client autodocumentation updated to current master branch

* Added movement functionality for the robots

The board can now move the player robots around

ghuser1023 committed:
Client abstraction layer (#63)

* change remotecontroller indentation style

* add abstraction layer

* changes from peer review

* fix bugs

reginawang3495 committed:
add to git ignore and merge with master

reginawang3495 committed:
fix CI test script

reginawang3495 committed:
update heroku api key

reginawang3495 reviewed (approved):
resolved conflicts and update api key for CD!
